modin/pandas/test/dataframe/test_binary.py

```diff
@@ -353,3 +354,17 @@ def test_add_custom_class():
         *create_test_dfs(test_data["int_data"]),
         lambda df: df + CustomIntegerForAddition(4),
     )
+
+
+def test_non_commutative_multiply():
+    # This test checks that mul and rmul do different things when
+    # multiplication is not commutative, e.g. for adding a string to a string.
+    # For context see https://github.com/modin-project/modin/issues/5238
+    modin_df, pandas_df = create_test_dfs([1], dtype=int)
+    integer = NonCommutativeMultiplyInteger(2)
+    # It's tricky to get the non commutative integer class implementation
+    # right, so before we do the actual test, check that the operation
+    # we care about is really not commmutative in pandas.
+    assert not (integer * pandas_df).equals(pandas_df * integer)
```

Comment:
this should be a separate test that implementation of `NonCommutativeMultiplyInteger` is correct

Reply:
Done.